posts/views.py

```diff
@@ -16,12 +17,150 @@ def post_create(request):
+    sorted_ingredient_counts = sorted(ingredient_counts.items(), key=operator.itemgetter(1))
+    sorted_ingredient_counts.reverse()
+    ingredients_list = []
+    for ingredient in sorted_ingredient_counts:
+        ingredients_list.append(Ingredients.objects.get(name=ingredient[0]))
+
+    for post in post_list:
+        num = 0
+        for rate in post.rate_set.all():
+            num += rate.rating
+        if post.rate_set.all():
+            average = num / post.rate_set.all().count()
+            post.rating = average
+        else:
+            post.rating = 0
+        post.save()
+
+    return render(request, 'posts/post_list.html', context={'post_list': post_list, 'ingredients': ingredients_list})
+
+
+def post_detail(request, pk):
+    post = Post.objects.get(pk=pk)
+    rate_form = RateForm()
+
+    if request.user.is_authenticated:
+        like = Like.objects.filter(post=post, user=request.user)
+        if Rate.objects.filter(post=post, user=request.user):
+            rate = Rate.objects.get(post=post, user=request.user)
+            rate_form = RateForm(instance=rate)
+    else:
+        like = None
+
+    return render(request, 'posts/post_detail.html', context={'post': post, 'like': like, 'rate_form': rate_form})
+
+
+def post_edit(request, pk):
+    if request.user.is_authenticated and request.method == 'POST':
+        post = Post.objects.get(pk=pk)
+        if post.owner == request.user:
+            post_form = PostForm(instance=post)
+            return render(request, 'posts/post_edit.html', context={'form': post_form, 'pk': pk})
+    return HttpResponseRedirect(reverse('posts:detail', kwargs={'pk': pk}))
+
+
+def update(request, pk):
+    if request.user.is_authenticated and request.method == 'POST':
+        post = Post.objects.get(pk=pk)
+
+        if post.owner == request.user:
+            form = PostForm(data=request.POST, instance=post)
+
+            if form.is_valid():
+                post_updated = form.save(commit=True)
+                post_updated.ingredients.clear()
+
+                for choice in post_updated.choices:
+                    if not Ingredients.objects.filter(name=choice):
+                        ingredient = Ingredients(name=choice)
+                        ingredient.save()
+                    post.ingredients.add(Ingredients.objects.get(name=choice))
+
+                return HttpResponseRedirect(reverse('posts:detail', kwargs={'pk': pk}))
+
+            return render(request, 'posts/post_edit.html', context={'form': form, 'pk': pk})
+
+    return HttpResponseRedirect(reverse('posts:detail', kwargs={'pk': pk}))
+
+
+def delete(request, pk):
+    if request.user.is_authenticated and request.method == 'POST':
+        post = Post.objects.get(pk=pk)
+
+        if post.owner == request.user:
+            post.delete()
+
+    return HttpResponseRedirect(reverse('posts:post_list'))
+
+
+def handle_like(request, pk):
+    if request.user.is_authenticated:
```

Comment:
Can we use `login-required` decorator in this view and in other required views ?
https://docs.djangoproject.com/en/2.2/topics/auth/default/#the-login-required-decorator